src/main/java/org/terasology/climateConditions/HyperthermiaSystem.java

```diff
@@ -97,17 +113,17 @@ public void onSpawn(OnPlayerSpawnedEvent event, EntityRef player, LocationCompon
         }
     }
 
-    private void weakenPlayer(EntityRef player, HealthComponent health) {
-        player.send(new ChangeMaxHealthEvent(newMaxHealth));
+    private void weakenPlayer(EntityRef player, HealthComponent health, HyperthermiaComponent hyperthermia) {
+        player.send(new ChangeMaxHealthEvent(hyperthermia.maxHealthMultiplier * health.maxHealth));
         health.currentHealth = Math.min(health.currentHealth, health.maxHealth);
-        health.regenRate *= regenReduceFactor;
+        health.regenRate *= hyperthermia.regenMultiplier;
         player.saveComponent(health);
     }
 
-    private void removePlayerWeakness(EntityRef player, HealthComponent health) {
+    private void removePlayerWeakness(EntityRef player, HealthComponent health, HyperthermiaComponent hyperthermia) {
```

Comment:
While we're at it, can we align `weakenPlayer` and `removePlayerWeakness` :sweat_smile:

Reply:
Yeah I sort of couldn't think of an opposite to weakenPlayer 😅, I mean strengthen Player sounds a little different than what the function does....
And addPlayerWeakness also sounded weird to me ..